rails.6.1.gemfile.lock

```diff
@@ -0,0 +1,282 @@
+PATH
+  remote: .
+  specs:
+    infinum_json_api_setup (0.0.8)
+      json_schemer (~> 0.2)
+      jsonapi-query_builder
+      jsonapi-serializer
+      jsonapi_parameters
+      pagy
+      rails
+      responders
+
+GEM
+  remote: https://rubygems.org/
+  specs:
+    actioncable (6.1.7)
```

Comment:
I'm not used to gem development, could you explain why we need a gemfile in this project? I was expecting just a gemspec

Reply:
the primary reason is, that's how [gemika](https://github.com/makandra/gemika) works. If you want to test your code against different dependency setups, then you have to have one Gemfile per dependency setup

Reply:
And what about before, could we have gone without it if we locked rails in the gemspec?

Reply:
Why do you want to get rid of Gemfile and Gemfile.lock?

> could we have gone without the Gemfile & Gemfile.lock if we locked rails in the gemspec instead?
Rails in gemspec is defined as a runtime dependency, if we lock it to a certain version, that would mean our gem is working with only one version of Rails

Reply:
I was taking a look at jsonapi_query_builder and this gem yesterday and noticed that they're setup differently, one has everything in the gemspec https://github.com/infinum/jsonapi-query_builder/blob/master/jsonapi-query_builder.gemspec, while this one had gems specified in both the gemfile and gemspec

> Rails in gemspec is defined as a runtime dependency, if we lock it to a certain version, that would mean our gem is compatible with only one version of Rails.

Agreed, to be more specific I wanted to ask "why do we need the gemfile.lock, couldn't we lock the rails version to >5, < 8 in the gemspec instead?"

I'll take some time and re-read https://yehudakatz.com/2010/12/16/clarifying-the-roles-of-the-gemspec-and-gemfile/, I'm not entirely clear on the purpose and suggested usages of these files

Reply:
👍 from my perspective, I don't think there is a practical difference between specifying development deps in gemspec vs Gemfile